components/automate-deployment/pkg/client/restore.go

```diff
@@ -196,65 +198,69 @@ func (r *DeploymentRestore) stopDeploymentService() error {
 		)
 	}
 
-	startTime := time.Now()
-	timeout := 10 * time.Second
 	sleepTime := 500 * time.Millisecond
-	ctx := context.Background()
+	doneChan := make(chan struct{})
 
 	// Wait for Habitat to "unload" the process
-	for {
-		deployed, err := r.target.DeployedServices(ctx)
-		if err == nil {
-			alive := false
-			for _, svc := range deployed {
-				if svc.Pkg.Name() == deploymentServiceName {
-					alive = true
+	go func() {
```

Comment:
[nit] I've been thinking we should prefer named functions for any `go func` statements that extend for more than a handful of lines.  It makes the goroutine a bit easier to spot in dlv and allows us to give a name to the thing we think this is doing.